src/core/Akka.Streams/Implementation/IO/OutputStreamSourceStage.cs

```diff
@@ -231,7 +283,14 @@ private void SendData(ByteString data)
         {
             Send(() =>
             {
-                _dataQueue.Add(data);
+                try
+                {
+                    _dataQueue.Add(data);
+                }
+                catch (Exception ex)
```

Comment:
I think, the canonical impl just didn't care about that. Personally I would vote for avoiding IOException inception.